allen-cell-animated/nucmorph-colorizer
# fix: UX updates to settings sliders

Problem
=======
Closes #211, implementing various fixes to the slider UI.

*Estimated review size: small-med, ~20 minutes. May be slightly easier to read w/o whitespace changes*

Solution
========
This change was bigger than expected because of a UX suggestion to add input boxes to the sliders. There's a lot of similar code in the `LabeledRangeSlider` component, so rather than copy the logic I opted to update it to handle both ranges and single inputs for DRY reasons.

- Refactored `LabeledSlider` (from `LabeledRangeSlider`) to handle range type inputs and single numeric value inputs based on input props.
- Adjustments to settings sliders:
  - Settings sliders now use `LabeledSlider` under the hood and have a numeric input box.
  - Increases brightness range from 50-150% to 0-200%
  - Fixed inconsistent use of `%` labels on percentages.
  - Sliders use consistent integer stepping
  - Sliders only have one marker, at default value

## Type of change

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-293/#/viewer?dataset=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Ftest%2F3500005828_25%2Fmanifest.json&feature=adjacent_neighbors&tab=settings&bg-key=ch0ch1&scatter-range=all
2. Go to the Viewer settings tab and adjust the values on the slider. You should see the viewer update as expected. You can also try typing into the input or giving it non-numeric/bad values.
3. Test the color ramp slider to ensure it still works as expected.

Screenshots (optional):
-----------------------
| Before | After |
| --- | --- |
| ![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/3fa41a72-6e14-42db-b224-29d37e39117f) | ![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/f6646d3a-30c0-4a74-8220-2c15d85cf43d) |


Keyfiles (delete if not relevant):
-----------------------
1. `LabeledSlider.tsx`
4. `SettingsTab.tsx`

Thanks for contributing!
